### ISSUE TITLE
 help. bs can not work when i set socket.namespace like this.

### ISSUE THREAD

**Issue by xunuo** [BrowserSync/browser-sync#621]
can not work when i set namespace like this.
````
socket: {
  namespace: function (namespace) {
    return "localhost:3000" + namespace;
  }
}
````
@version 2.7.1

**Comment by shinnn** [comment]
> can not work

https://github.com/BrowserSync/browser-sync/blob/03123954b6585a63ad53e7d8f730430a6a60b6ee/CONTRIBUTING.md#-tips-for-a-creating-a-great-issue-report

**Comment by xunuo** [comment]
````
gulp.task('default', function() {
    browserSync.init({
    	host: "192.168.1.110",
    	files: "*.html",
        server: {
        	baseDir: "./",
	        directory: true,
	        host: "192.168.1.110"
        },
        open: false,
        scriptPath: function (path, port, options) {
		    return options.get("absolute").replace('HOST','192.168.1.110');
		},
        socket: {
        	path : '/ooooo.io',
		    clientPath : '/short',
		    // issue: bs can not work when i set namespace like this.
		    namespace: function (namespace) {
		        return "localhost:3000" + namespace;
		    }
		}
    });
});
````

**Comment by xunuo** [comment]
hello

**Comment by shinnn** [comment]
What do you mean by "cannot work"?

**Comment by xunuo** [comment]
Hello, that's mean 'no browser-sync layer appear at top right corner'.

**Comment by xunuo** [comment]
hello,help.

**Comment by shinnn** [comment]
@shakyShane Thoughts?

**Comment by shakyShane** [comment]
A fix will land soon https://github.com/BrowserSync/browser-sync/commit/51574325a716712c0b53725575f268f1f44b7d8b

**Comment by shakyShane** [comment]
Fixed in `v2.7.13`

The original implementation was completely broken (where setting the namespace allowed a url, doh!).

now, to achieve what you want, it's the domain property you want.

```js
socket: {
    domain: "localhost:3000"
}
```

please also see https://github.com/BrowserSync/browser-sync/issues/690
